File: algo/1. Two Sum.js
// @9 Two Sum
// O(n) time complexity if using hashmap

/*
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        nums_dict = {}
        for i, num in enumerate(nums):
            nums_dict[num] = i
        for i, num in enumerate(nums):
            if (target-num) in nums_dict.keys():
                if i != nums_dict[target-num]:
                    return [i, nums_dict[target-num]]
*/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    for (var i = 0; i < nums.length; i++) {
        for (var j=i+1; j< nums.length; j++) {
            if (nums[i]+nums[j] == target) {
                return [i,j];
            }
        }
    }
    
    return null;
};

File: round3/1. Two Sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var res = new int[2];
        
        var dict = new Dictionary<int,int>();
        for (int i = 0; i < nums.Length; i ++) {
            dict[nums[i]] = i;
        }
        
        for (int i = 0; i < nums.Length; i ++) {
            if (dict.ContainsKey(target - nums[i])) {
                int temp = dict[target-nums[i]];
                if (temp == i) continue;
                res[0] = i;
                res[1] = temp;
                return res;
            }
        }
        
        return res;
    }
}